drift/lib/src/sentry_query_interceptor.dart

```diff
@@ -0,0 +1,164 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import 'dart:async';
+
+import 'package:drift/drift.dart';
+import 'package:meta/meta.dart';
+import 'package:sentry/sentry.dart';
+
+import 'constants.dart' as drift_constants;
+import 'sentry_span_helper.dart';
+import 'version.dart';
+
+/// A Sentry query interceptor that wraps database operations in performance monitoring spans.
+///
+/// This interceptor tracks all database operations executed through a Drift database connection,
+/// including transactions, batches, and individual CRUD operations. Each operation is captured
+/// as a Sentry span with relevant context.
+class SentryQueryInterceptor extends QueryInterceptor {
+  final String _dbName;
+  late final SentrySpanHelper _spanHelper;
+  bool _isDbOpen = false;
+
+  SentryQueryInterceptor({required String databaseName, @internal Hub? hub})
+      : _dbName = databaseName {
+    _spanHelper = SentrySpanHelper(
+      SentryTraceOrigins.autoDbDriftQueryInterceptor,
+      hub: hub,
+    );
+    final options = hub?.options;
```

Comment:
Don't we need to fallback to `HubAdapter`? We do this in SentrySpanHelper, but nut here. Wenn we call the class like in the example, options will be null here, unless i'm missing something.

```dart
SentryQueryInterceptor(databaseName: 'your_db_name'),
```

Reply:
thx for noticing, the tests didn't catch that because we pass the hub directly

will correct it